workflows/runSh.js

```diff
@@ -23,6 +23,10 @@ var cleanup = require('../job/cleanup.js');
 var updateStatus = require('../job/updateStatus.js');
 
 function runSh(externalBag, callback) {
+  // At this point we have started processing the runSh and we do not want
+  // the cluster node validation to affect the build.
+  global.config.isProcessingRunSh = true;
```

Comment:
This deviates from the original design, but is a much simpler way of figuring out if the node is processing a job currently.

Reply:
can we just call it `isJobInProgress` or `isProcessingJob` to be more clear on what this is

Reply:
I didn't want it to be confused with a CI job.

Reply:
`isProcessingRunShJob` maybe?